src/App.tsx

```diff
@@ -1,34 +1,234 @@
-/* eslint-disable max-len */
-import React from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import React from 'react';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
-export const App: React.FC = () => {
-  return (
-    <>
-      <div className="section">
-        <div className="container">
-          <div className="box">
-            <h1 className="title">Todos:</h1>
+interface State {
+  todos: Todo[];
+  queryTodos: Todo[];
+  loading: boolean;
+  todoId: number | string;
+  query: string;
+  filter: string;
+}
 
-            <div className="block">
-              <TodoFilter />
-            </div>
+export class App extends React.Component<{}, State> {
+  state:State = {
+    todos: [],
+    queryTodos: [],
+    loading: true,
+    todoId: 0,
+    query: '',
+    filter: '',
+  };
+
+  componentDidMount() {
+    getTodos()
+      .then(todos => {
+        this.setState({
+          todos,
+          queryTodos: todos,
+          loading: false,
+        });
+      });
+  }
+
+  handleChangeAll = () => {
+    this.setState(prevState => {
+      return {
+        queryTodos: prevState.todos,
+        filter: '',
+      };
+    });
+
+    if (this.state.query) {
+      this.setState(prevState => {
+        const allQuery = prevState.todos.filter(todo => todo.title.toLowerCase()
+          .includes(this.state.query.toLowerCase().trim()));
+
+        return {
+          queryTodos: allQuery,
+          filter: '',
+        };
+      });
+    }
+  };
+
+  handleChangeCompleted = () => {
+    this.setState(prevState => {
+      const completedTodos
+    = prevState.todos.filter(todo => todo.completed);
```

Comment:
```suggestion
      const completedTodos = prevState.todos
        .filter(todo => todo.completed);
```
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/96409563/f6f81584-f2a2-4ed8-9179-6777bb9bfdb0)